ray-operator/controllers/ray/rayservice_controller.go

```diff
@@ -960,12 +956,27 @@ func (r *RayServiceReconciler) getAndCheckServeStatus(ctx context.Context, dashb
 	return isReady, nil
 }
 
-func (r *RayServiceReconciler) generateConfigKey(rayServiceInstance *rayv1.RayService, clusterName string) string {
-	return r.generateConfigKeyPrefix(rayServiceInstance) + clusterName
+func (r *RayServiceReconciler) getServeConfigFromCache(rayServiceInstance *rayv1.RayService, clusterName string) string {
+	cacheKey := rayServiceInstance.Namespace + "/" + rayServiceInstance.Name
+	serveConfigs, exist := r.ServeConfigs.Get(cacheKey)
+	if !exist {
+		return ""
```

Comment:
Is it possible that the config is indeed an empty string?

Reply:
Fixed in https://github.com/ray-project/kuberay/pull/2591/commits/040d4755b0ba54b48130c35456b72a5520ccc4e0. Only cache the serveConfig when it is not empty.